syntax = "proto3";

enum Type {
    IMAGE = 0;
    VIDEO = 1;
    VIDEO_LOOPING = 2;
    VIDEO_LOOPING_RANDOM = 3;
    GIF = 4;
}

message Artist {
    string uri = 1;
    string name = 2;
    string avatar = 3;
}

message Thumbnail {
    uint64 height = 1;
    uint64 width = 2;
    string url = 3;
}

message EntityCanvazResponse {
    repeated Canvaz canvases = 1;
    message Canvaz {
        string id = 1;
        string url = 2;
        string file_id = 3;
        Type type = 4;
        string entity_uri = 5;
        Artist artist = 6;
        bool explicit = 7;
        string uploaded_by = 8;
        string etag = 9;
        string canvas_uri = 11;
        string storylines_id = 12;
        repeated Thumbnail thumbnails = 13;
    }
    int64 ttl_in_seconds = 2;
}

message EntityCanvazRequest {
    repeated Entity entities = 1;
    message Entity {
        string entity_uri = 1;
        string etag = 2;
    }
}
